cmd/runsim/integrations/slack/main.go

```diff
@@ -119,151 +78,142 @@ func processSlackRequest(slashCmdPayload string) (CircleApiPayload, error) {
 			for _, param := range simParams {
 				switch param {
 				case reBlocks.FindString(param):
-					circleApiPayload.BuildParameters.Blocks = param
+					payload.BuildParameters.Blocks = param
 				case "yes":
-					circleApiPayload.BuildParameters.Genesis = "true"
+					payload.BuildParameters.Genesis = "true"
 				// Without this case the branch will be set to "no"
 				// Temporary solution until we figure out how to handle custom genesis better
 				// TODO: Figure out a better way to handle genesis file
 				case "no":
-					circleApiPayload.BuildParameters.Genesis = "false"
+					payload.BuildParameters.Genesis = "false"
 				default:
-					circleApiPayload.BuildParameters.CommitHash = param
+					payload.BuildParameters.CommitHash = param
 				}
 			}
 			continue
 		}
-		circleApiPayload.BuildParameters.SlackResponseUrl = match
+		payload.BuildParameters.IntegrationType = "slack"
+		respUrl = match
 	}
-	return *circleApiPayload, nil
+	return
 }
 
-func verifySlackRequest(slackSig, slackTimestamp, requestBody string) error {
-	var slackSecret string
-	var intTimestamp int64
-
-	if intTimestamp, err = strconv.ParseInt(slackTimestamp, 10, 64); err != nil {
-		return err
-	}
-	if slackSecret, err = getSsmParameter(paramNameSlackSecret); err != nil {
-		return err
+func verifySlackRequest(slackSig, slackTimestamp, slackSecret, requestBody string) (err error) {
+	intTimestamp, err := strconv.ParseInt(slackTimestamp, 10, 64)
+	if err != nil {
+		return
 	}
+
 	if math.Abs(time.Since(time.Unix(intTimestamp, 0)).Seconds()) > 120 {
 		// The request timestamp is more than two minutes from local time.
 		// It could be a replay attack, so let's ignore it.
 		return errors.New("timestamp check failed")
 	}
 
 	requestHash := hmac.New(sha256.New, []byte(slackSecret))
-	if _, err = requestHash.Write([]byte("v0:" + slackTimestamp + ":" + requestBody)); err != nil {
-		return err
+	if _, err = requestHash.Write([]byte(fmt.Sprintf("v0:%s:%s", slackTimestamp, requestBody))); err != nil {
+		return
 	}
 	if hmac.Equal([]byte(slackSig), []byte("v0="+hex.EncodeToString(requestHash.Sum(nil)))) {
-		return nil
+		return
 	}
 	return errors.New("failed")
 }
 
-func circleciStatusCheck() error {
-	var data []CircleStatusCheckResp
-
-	circleUrl := "https://circleci.com/api/v1.1/project/github/tendermint/images?limit=1&shallow=true"
-	if request, err = http.NewRequest("GET", circleUrl, nil); err != nil {
-		return err
+func triggerCircleciJob(circleToken string, payload CircleApiPayload) (err error) {
```

Comment:
this can be moved to a common package to avoid dups

Reply:
Would this break the build until the common package is merged to master?